sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/CompositePath.java

```diff
@@ -26,7 +28,7 @@ public CompositePath() {
      * @param jsonString the json string that represents the included path.
      */
     CompositePath(String jsonString) {
-        super(jsonString);
+        this.jsonSerializable = new JsonSerializable(jsonString);
```

Comment:
I am not sure yet if we should have JsonSerializableWrapper, but if we decide to go with JsonSerializable, then you could move this code to JsonSerializable and and here just have super(jsonStirng).

Reply:
yup, good idea to move the common code into the parent class

Reply:
based on the discussion, will remove JsonSerializableWrapper and ResourceWrapper in  following up PR.